mlir/lib/Quantum/Transforms/ChainedSelfInversePatterns.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define DEBUG_TYPE "chained-self-inverse"
+
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/Errc.h"
+
+#include "Quantum/IR/QuantumOps.h"
+#include "Quantum/Transforms/Patterns.h"
+
+using llvm::dbgs;
+using namespace mlir;
+using namespace catalyst;
+using namespace catalyst::quantum;
+
+namespace {
+
+llvm::SmallVector<StringRef> SelfInverseGates{
+    "PauliX", "PauliY", "PauliZ", "Hadamard", "CNOT", "CY", "CZ", "SWAP", "Toffoli",
```

Comment:
Awesome! Maybe we should order them alphabetically?

Reply:
Good idea, I also grouped them based on number of bits so it's clearer, instead of just a long chunk of string texts.